lexicon/tests/providers/test_namecom.py

```diff
@@ -0,0 +1,148 @@
+"""Integration tests for Name.com"""
+import json
+from unittest import TestCase
+
+import pytest
+from mock import ANY, Mock, patch
+from requests import HTTPError
+
+from lexicon.config import DictConfigSource
+from lexicon.providers.namecom import provider_parser
+from lexicon.tests.providers.integration_tests import (
+    IntegrationTests, _vcr_integration_test
+)
+
+
+# Hook into testing framework by inheriting unittest.TestCase and reuse
+# the tests which *each and every* implementation of the interface must
+# pass, by inheritance from integration_tests.IntegrationTests
+class NamecomProviderTests(TestCase, IntegrationTests):
+    """TestCase for Name.com"""
+
+    # I don't think we really need some docstrings here.
+    # pylint: disable=missing-function-docstring
+
+    provider_name = 'namecom'
+    domain = 'mim.pw'
+
+    def _filter_headers(self):
+        return ['Authorization', 'Cookie']
+
+    def _filter_response(self, response):
+        headers = response['headers']
+        headers.pop('Set-Cookie', None)
+        headers.pop('content-length', None)
+
+        if response['status']['code'] == 200:
+            try:
+                data = json.loads(response['body']['string'].decode())
+            except ValueError:
+                pass
+            else:
+                if 'records' in data:
+                    min_id = 10 ** 8
+                    data['records'] = [
+                        record for record in data['records']
+                        if record['id'] > min_id
+                    ]
+                    response['body']['string'] = json.dumps(data).encode()
+
+        return response
+
+    ###########################
+    # Provider.authenticate() #
+    ###########################
+    @_vcr_integration_test
+    def test_provider_authenticate(self):
```

Comment:
Could you rename this test to not shadow the existing one in `integration_test`?